src/Workspaces/MSBuildTest/OpenProjectsTests.cs

```diff
@@ -0,0 +1,222 @@
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled))]
+        [InlineData("api")]
+        [InlineData("web")]
+        [InlineData("grpc")]
+        [InlineData("webapi")]
+        [InlineData("webapp")]
+        [InlineData("mvc")]
+        [InlineData("angular")]
+        [InlineData("react")]
+        [InlineData("blazor")]
+        [InlineData("blazorwasm")]
+        [InlineData("classlib")]
+        [InlineData("console")]
+        [InlineData("mstest")]
+        [InlineData("nunit")]
+        [InlineData("razorclasslib")]
+        [InlineData("worker")]
+        [InlineData("xunit")]
+        public async Task CSharpTemplateProject_LoadWithNoDiagnostics(string templateName)
+        {
+            var ignoredDiagnostics = templateName switch
+            {
+                _ => Array.Empty<string>(),
+            };
+
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.CSharp, ignoredDiagnostics);
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled), typeof(WindowsOnly))]
+        [InlineData("winforms")]
+        [InlineData("winformslib")]
+        [InlineData("winformscontrollib")]
+        [InlineData("wpf")]
+        [InlineData("wpflib")]
+        [InlineData("wpfcustomcontrollib")]
+        [InlineData("wpfusercontrollib")]
+        public async Task VisualBasicTemplateProject_WindowsOnly_LoadWithNoDiagnostics(string templateName)
+        {
+            var ignoredDiagnostics = templateName switch
+            {
+                _ => Array.Empty<string>(),
+            };
+
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.VisualBasic, ignoredDiagnostics);
+        }
+
+        [ConditionalTheory(typeof(DotNetSdkMSBuildInstalled))]
+        [InlineData("classlib")]
+        [InlineData("console")]
+        [InlineData("mstest")]
+        [InlineData("nunit")]
+        [InlineData("xunit")]
+        public async Task VisualBasicTemplateProject_LoadWithNoDiagnostics(string templateName)
+        {
+            var ignoredDiagnostics = (templateName, isWindows: ExecutionConditionUtil.IsWindows) switch
+            {
+                (_, isWindows: false) => new string[] { ERR_UndefinedType1 },
+                _ => Array.Empty<string>(),
+            };
+
+            await AssertTemplateProjectLoadsCleanlyAsync(templateName, LanguageNames.VisualBasic, ignoredDiagnostics);
+        }
+
+        private async Task AssertTemplateProjectLoadsCleanlyAsync(string templateName, string languageName, string[] ignoredDiagnostics)
+        {
+            try
+            {
+                if (ignoredDiagnostics.Length > 0)
+                {
+                    TestOutputHelper.WriteLine($"Ignoring compiler diagnostics: \"{string.Join("\", \"", ignoredDiagnostics)}\"");
+                }
+
+                // Clean up previous run
+                CleanupProject(templateName, languageName);
```

Comment:
We have temporary file helpers in our core test helpers; should we use that?

Reply:
Investigated and SolutionDirectory is actually a `DisposableDirectory` now use it as such.